src/main.rs

```diff
@@ -103,205 +109,215 @@ pub enum RunAsRole {
-                        set as controller, node, async-fuse or scheduler-extender, \
-                        default as node",
-                ),
-        )
-        .arg(
-            Arg::with_name(ETCD_ADDRESS_ARG_NAME)
-                .short("e")
-                .long(ETCD_ADDRESS_ARG_NAME)
-                .value_name("ETCD IP:PORT,ETCD IP:PORT")
-                .takes_value(true)
-                .required(true)
-                .help(
-                    "Set the etcd addresses of format ip:port, \
-                        if multiple etcd addresses use comma to seperate, \
-                        required argument, no default value",
-                ),
-        )
-        .arg(
-            Arg::with_name(SCHEDULER_EXTENDER_PORT_ARG_NAME)
-                .long(SCHEDULER_EXTENDER_PORT_ARG_NAME)
-                .value_name("SCHEDULER EXTENDER PORT")
-                .takes_value(true)
-                .help("Set the port of the scheduler extender"),
-        )
-        .arg(
-            clap::Arg::with_name(VOLUME_INFO_ARG_NAME)
-                .short("v")
-                .long(VOLUME_INFO_ARG_NAME)
-                .value_name("VOLUME_INFO")
-                .takes_value(true)
-                .required(true)
-                .help(
-                    "Set volume backend information, \
-                        required argument, no default value",
-                ),
-        )
-        .arg(
-            clap::Arg::with_name(VOLUME_TYPE_ARG_NAME)
-                .long(VOLUME_TYPE_ARG_NAME)
-                .value_name("VOLUME_TYPE")
-                .takes_value(true)
-                .help(
-                    "Set volume backend type, \
-                        required argument",
-                ),
-        )
-        .arg(
-            clap::Arg::with_name(SERVER_PORT_NUM_ARG_NAME)
-                .long(SERVER_PORT_NUM_ARG_NAME)
-                .value_name("PORT_NUM")
-                .takes_value(true)
-                .help(&format!(
-                    "Set service port number, \
-                        required argument, default value is {}",
-                    DEFAULT_PORT_NUM
-                )),
-        )
-        .get_matches();
-    get_args(&matches)
+impl ArgParam {
+    /// Get the arg name for index
+    #[must_use]
+    const fn get_name(&self) -> &'static str {
+        self.name
+    }
+
+    /// Generate the arg with short&long or not
+    #[must_use]
+    fn new_arg<'a>(&self) -> Arg<'a, 'a> {
+        let mut arg = Arg::with_name(self.name)
+            .value_name(self.value_name)
+            .takes_value(self.take_value)
+            .required(self.required)
+            .help(self.help);
+
+        if let Some(s) = self.short {
+            arg = arg.short(s);
+        }
+
+        if let Some(l) = self.long {
+            arg = arg.long(l);
+        }
+
+        arg
+    }
+}
+
+/// Generate the default arg
+//allow for this function as there is no other logic in this function
```

Comment:
put this comment after #[allow(clippy::too_many_lines)]

Reply:
done